SDK/AppCenterCrashes/Microsoft.AppCenter.Crashes.Windows.Shared/Crashes.cs

```diff
@@ -117,20 +112,44 @@ internal Crashes()
         /// </summary>
         internal Task ProcessPendingErrorsTask { get; set; }
 
+        // Task to get the last session error report, if one is found.
+        private Task<ErrorReport> _lastSessionErrorReportTask;
+
+        internal Crashes()
+        {
+            _unprocessedManagedErrorLogs = new Dictionary<Guid, ManagedErrorLog>();
+        }
+
         /// <summary>
         /// Method that is called to signal start of Crashes service.
         /// </summary>
         /// <param name="channelGroup">Channel group</param>
         /// <param name="appSecret">App secret</param>
         public override void OnChannelGroupReady(IChannelGroup channelGroup, string appSecret)
         {
-            lock (CrashesLock)
+            lock (_serviceLock)
             {
                 base.OnChannelGroupReady(channelGroup, appSecret);
                 ApplyEnabledState(InstanceEnabled);
                 if (InstanceEnabled)
                 {
                     ProcessPendingErrorsTask = ProcessPendingErrorsAsync();
+                    _lastSessionErrorReportTask = Task.Run(() =>
```

Comment:
Should this be in `ApplyEnabledState` and set `null` when it is disabled?

Reply:
Modified to set null when disabled, but kept it in onchannelgroupready as an optimization.